src/main/java/com/buglife/sdk/FileAttachment.java

```
@@ -0,0 +1,115 @@
+ * limitations under the License.
+ *
+ */
+
+package com.buglife.sdk;
+
+import android.os.Parcel;
+import android.os.Parcelable;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.Base64;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.Serializable;
+
+public class FileAttachment implements Parcelable {
+    @NonNull private final File mFile;
+    @NonNull private final String mMimeType;
+
+    public FileAttachment(@NonNull File file, @NonNull String mimeType) {
+        this.mFile = file;
+        this.mMimeType = mimeType;
+    }
+
+    public JSONObject toJSON() throws JSONException {
+        JSONObject json = new JSONObject();
+        json.put("filename", mFile.getName());
+
+        byte[] data = toByteArray();
+        if (data != null) {
+            json.put("base64_attachment_data", Base64.encodeToString(data, Base64.NO_WRAP));
+        }
+
+        json.put("mime_type", mMimeType);
+        return json;
+    }
+
+    @NonNull public File getFile() {
+        return mFile;
+    }
+
+    public boolean isImage() {
+        return mMimeType.equals(MimeTypes.JPG) || mMimeType.equals(MimeTypes.PNG);
+    }
+
+    public boolean isVideo() {
+        return mMimeType.equals(MimeTypes.MP4);
+    }
+
+    @Nullable
+    private byte[] toByteArray() {
+        FileInputStream inputStream = null;
+        ByteArrayOutputStream outputStream = null;
+        try {
+            inputStream = new FileInputStream(mFile);
+            outputStream = new ByteArrayOutputStream((int) mFile.length());
+            IOUtils.write(inputStream, outputStream);
+            return outputStream.toByteArray();
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
```

Seems like we should bubble up these exceptions? `toJSON()` already throws ( ¯\_(ツ)_/¯)